fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -4996,10 +5001,9 @@ class FMCMainDisplay extends BaseAirliners {
     //TODO: Can this be util?
     getGW() {
         let fmGW = 0;
-        if (this.isAnEngineOn() && isFinite(this.zeroFuelWeight)) {
+        //Simplified to just checking fuelWeight as GetFOB handles what fuel level to use (block vs tank reading)
+        if (isFinite(this.zeroFuelWeight) && this.blockFuel) {
```

Comment:
I don't think checking for `this.blockFuel` works here. With an engine running, the gross weight should be available (since fuel is available from FQI) even if no block fuel is entered.

What I suggest is fetching `this.getFOB()` and checking whether that's finite instead.

Also, I prefer `Number.isFinite` over `isFinite` since `isFinite(null)` is true which is rarely desired.

Reply:
//Does this work?  I removed unnecessary else block for simplicity sake as well since fmGw already is at 0.


`getGW() {
        let fmGW = 0;
        const currFob = this.getFOB();
        //Simplified to just checking fuelWeight as GetFOB handles what fuel level to use (block vs tank reading)
        if (Number.isFinite(this.zeroFuelWeight) && Number.isFinite(currFob)) {
            fmGW = (currFob + this.zeroFuelWeight);
        }
        SimVar.SetSimVarValue("L:A32NX_FM_GROSS_WEIGHT", "Number", fmGW);
        return fmGW;
    }`